src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -65,5 +66,92 @@ public static void BuildChainExtraStoreUntrustedRoot()
             }
         }
+
+        public static IEnumerable<object[]> VerifyExpressionData()
+        {
+            // The test will be using the chain for TestData.MicrosoftDotComSslCertBytes
+            // The leaf cert (microsoft.com) is valid from 2014-10-15 00:00:00Z to 2016-10-15 23:59:59Z
+            DateTime[] validTimes =
+            {
+                // The NotBefore value
+                new DateTime(2014, 10, 15, 0, 0, 0, DateTimeKind.Utc),
+
+                // One second before the NotAfter value
+                new DateTime(2016, 10, 15, 23, 59, 58, DateTimeKind.Utc),
+            };
+
+            // The NotAfter value as a boundary condition differs on Windows and OpenSSL.
+            // Windows considers it valid (<= NotAfter).
+            // OpenSSL considers it invalid (< NotAfter), with a comment along the lines of
+            //   "it'll be invalid in a millisecond, why bother with the <="
+            // So that boundary condition is not being tested.
+
+            DateTime[] invalidTimes =
+            {
+                // One second before the NotBefore time
+                new DateTime(2014, 10, 14, 23, 59, 59, DateTimeKind.Utc),
+
+                // One second after the NotAfter time
+                new DateTime(2016, 10, 16, 0, 0, 0, DateTimeKind.Utc),
+            };
+
+            List<object[]> testCases = new List<object[]>((validTimes.Length + invalidTimes.Length) * 3);
+
+            // Build (date, result, kind) tuples.  The kind is used to help describe the test case.
+            // The DateTime format that xunit uses does show a difference in the DateTime itself, but
+            // having the Kind be a separate parameter just helps.
+
+            foreach (DateTime utcTime in validTimes)
+            {
+                DateTime local = utcTime.ToLocalTime();
+                DateTime unspecified = new DateTime(local.Ticks);
+
+                testCases.Add(new object[] { utcTime, true, utcTime.Kind });
+                testCases.Add(new object[] { local, true, local.Kind });
+                testCases.Add(new object[] { unspecified, true, unspecified.Kind });
+            }
+
+            foreach (DateTime utcTime in invalidTimes)
+            {
+                DateTime local = utcTime.ToLocalTime();
+                DateTime unspecified = new DateTime(local.Ticks);
+
+                testCases.Add(new object[] { utcTime, false, utcTime.Kind });
+                testCases.Add(new object[] { local, false, local.Kind });
+                testCases.Add(new object[] { unspecified, false, unspecified.Kind });
+            }
+
+            return testCases;
+        }
+
+        [Theory]
+        [MemberData("VerifyExpressionData")]
+        public static void VerifyExpiration_LocalTime(DateTime verificationTime, bool shouldBeValid, DateTimeKind kind)
+        {
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
```

Comment:
This particular chain is already used for the basic sanity test, the DER bytes are at https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.X509Certificates/tests/TestData.cs#L944